### PR TITLE
Fixes #3930 Greeting text on home page to single textview

### LINKED ISSUE #3930: Greeting text on home screen should be a single TextView
For localization, it's hard to make sure that the reversed text for each of the two TextViews that comprise the home screen greeting correctly match the flipped layout that Android forms. This is *mostly* correct today, but it would be more correct if we could combine the strings together. The main challenge has been making sure that:
1. That the line properly extends the whole length of the TextView
2. That when the profile name is really long or display text is large, that the name properly wraps to the next line & the underline is properly resized to not extend past the last character

Another issue with current code is that the screen-reader reads it as `Good Morning,, Rajat!`. Basically there is one extra `,`. This is because the screen-reader adds `,` between two textviews and we already have one `,` in text itself.

### PR DESCRIPTION
**Pull request by yashbansal130**
Fixes #3930 

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by yashbansal130** [comment]
@yash10019coder look into it please

File: app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt [hunk 1]
```diff
@@ -208,83 +208,104 @@ class HomeActivityTest {
...
   @Test
-  fun testHomeActivity_eveningTimestamp_goodEveningMessageIsDisplayed() {
+  fun testHomeActivity_withEvening_AdminProfile_configChange_EveningWithProfileNameIsDisplayed() {
     fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_FIXED_FAKE_TIME)
     fakeOppiaClock.setCurrentTimeToSameDateTime(EVENING_TIMESTAMP)
-    launch<HomeActivity>(createHomeActivityIntent(internalProfileId1)).use {
+    launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
       testCoroutineDispatchers.runCurrent()
+      onView(isRoot()).perform(orientationLandscape())
       scrollToPosition(position = 0)
```
Comment (yash10019coder) [inline app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:307]: a testCouroutinesDispachers should be used before this line as we are rotating the activity which will destroy the activity and while it is being recreated it'll take some time

**Review by yash10019coder — CHANGES_REQUESTED** [review]
Hi @yashbansal130 left some comments PTAL thanks

**Comment by yash10019coder** [comment]
@rt4914  can you please approve the workflow

File: app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt (hunk 1 above)
Comment (yashbansal130) [reply app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:304]: does the above line (testCoroutineDispatchers.runCurrent()) does not do that, and if not can u provide me with an example of how to do this

Comment (yashbansal130) [reply app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:307]: > a testCouroutinesDispachers should be used before this line as we are rotating the activity which will destroy the activity and while it is being recreated it'll take some time

does the above line (testCoroutineDispatchers.runCurrent()) does not do that, and if not can u provide me with an example of how to do this

Comment (yash10019coder) [reply app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:307]: it can do that but it waits for the first time i.e. loading of the activity but we need it again as we are rotating the device

Comment (yashbansal130) [reply app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:307]: > it can do that but it waits for the first time i.e. loading of the activity but we need it again as we are rotating the device

can you provide me with an example code

Comment (yash10019coder) [reply app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:307]: image
like in this test

Comment (yashbansal130) [reply app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:307]: > image like in this test

I am also using a testCoroutineDispatchers.runCurrent() a line before changing orientation like in provided test example

Comment (yash10019coder) [reply app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:307]: I haven't written this test but I also prefer to use it after a orientation change so that it can give views time to load

Comment (yashbansal130) [reply app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:307]: > I haven't written this test but I also prefer to use it after a orientation change so that it can give views time to load

ok I will do it for all the tests which include orientation change

File: app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt [hunk 2]
```diff
@@ -208,83 +208,107 @@ class HomeActivityTest {
...
   }
 
   @Test
-  fun testHomeActivity_withAdminProfile_configChange_profileNameIsDisplayed() {
+  fun testHomeActivity_withAfternoonTime_withAdminProfile_afternoonWithProfileNameIsDisplayed() {
+    fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_FIXED_FAKE_TIME)
+    fakeOppiaClock.setCurrentTimeToSameDateTime(AFTERNOON_TIMESTAMP)
     launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
       testCoroutineDispatchers.runCurrent()
-      onView(isRoot()).perform(orientationLandscape())
```
Comment (rishidyno) [inline app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:237]: @yashbansal130 Why did you remove this line?

File: app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt [hunk 3]
```diff
@@ -208,83 +208,107 @@ class HomeActivityTest {
...
-        targetViewId = R.id.profile_name_text_view,
-        stringToMatch = "Admin!"
+        targetViewId = R.id.welcome_text_view,
+        stringToMatch = "Good morning,Admin!"
       )
     }
   }
 
   @Test
-  fun testHomeActivity_hasCorrectActivityLabel() {
```
Comment (rishidyno) [inline app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:224]: @yashbansal130 Also why did you remove this `testHomeActivity_hasCorrectActivityLabel` test. Please add it again as it tests activity label and create a separate test for `testHomeActivity_withMorningAndAdminProfile_configChange_MorningWithProfileNameIsDisplayed`.

**Review by rishidyno — CHANGES_REQUESTED** [review]

File: app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt (hunk 3 above)
Comment (yashbansal130) [reply app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:224]: it was removed by mistake, I will restore it.

File: app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt (hunk 2 above)
Comment (yashbansal130) [reply app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:237]: > @yashbansal130 Why did you remove this line?

I have added separate tests for each greetings message with and without config changes.

File: app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt (hunk 3 above)
Comment (yashbansal130) [reply app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:224]: > @yashbansal130 Also why did you remove this `testHomeActivity_hasCorrectActivityLabel` test. Please add it again as it tests activity label and create a separate test for `testHomeActivity_withMorningAndAdminProfile_configChange_MorningWithProfileNameIsDisplayed`.

@rishidyno it is not removed it is is shifted from this line, go to line number 211

Comment (rishidyno) [reply app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt:224]: @yashbansal130 please try not to shift the existing code unless needed as it makes code review a little confusing, and we have to do extra work to figure out the changes.

**Review by rishidyno — CHANGES_REQUESTED** [review]
@yashbansal130 PR almost looks good to me. Just fix failing lint test and also fix the failing tests `testHomeActivity_initialBrazilianPortugueseContext_displayStringsInPortuguese` and `testHomeActivity_initialArabicContext_displaysStringsInArabic` and similar if any.
